### DISCUSSION TITLE
Can you make multiple decks in one note using heirachical tags?

### DISCUSSION THREAD

**Discussion by riabac** [st3v3nmw/obsidian-spaced-repetition#744] [Q&A] [answered]
As title

**Comment by ronzulu** [comment, marked as answer]
Yes, by putting the tag at the start of the question text, e.g.

```
#flashcards Q1::A1

#flashcards Q2::A2
<!--SR:!2023-09-02,4,270-->

#flashcards Q3::A3
#flashcards/science Q4::A4 <!--SR:!2023-09-02,4,270-->
#flashcards/science/physics Q5::A5 <!--SR:!2023-09-02,4,270-->
#flashcards/math Q6::A6
```

The following might also be of interest:
https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/691

**Reply by riabac** [reply]
Great, thank you!

**Comment by Lucas256-cmd** [comment]
I think it would be helpful if a tag was valid until the next tag.

**Comment by ronzulu** [comment]
I agree.

There is an aspect of functionality to decide. 

### Treatment of question specific topics

Should what is currently a "question specific topic" be used even for later questions (that don't have a question specific topic)?

For example:
```
Q0::A0

#flashcards/deck1
Q1::A1

#flashcards/deck2 
Q2::A2

Q3::A3

#flashcards/deck3 Q4::A4

Q5::A5
```

Should the `#flashcards/deck3` that is on the same line as Q4 (and currently treated as a question specific tag) apply to Q5?

I.e. should Q5 be in topic:
| Topic | Reason |
| - | - |
| `#flashcards/deck3` | Option 1. It's the last topic tag prior to Q5 |
| `#flashcards/deck2` | Option 2. It's the last (non-question specific) topic tag prior to Q5 |

If this was all new code, then I'd suggest option 1, as it's easier to explain to users.

But option 2 would be more consistent with the current formula.

### Questions prior to the first topic tag

Current functionality: These questions have the topic of the first topic tag found (that is non-question specific)

I.e. in the above example, Q0 could be in topic  `#flashcards/deck1`

I think this seems reasonable. I think it should continue, modified if we choose option 1.

### Notes

For clarity, in the above example, the proposed change would mean that both Q2 and Q3 is in topic `#flashcards/deck2`



Cheers
Ronny

**Reply by suckerSlayer** [reply]
This would be great if be implemented. Adding a tag at every beginning of a line make it cluttered.

**Reply by Lucas256-cmd** [reply]
I think option 2 is the more logical one. And I think it gives the users more functionality too.
Oh and sorry for my late reply. I must have missed a notification or email about your answer. It is thanks to suckerSlayers comment I saw your answer :)

**Comment by ronzulu** [comment]
Hi all, this has been implemented and is available for beta testing at:
[PR: Support multi-deck cards #495, and multiple decks in a single note #705](https://github.com/st3v3nmw/obsidian-spaced-repetition/pull/834)

And @Lucas256-cmd I did implement option 2

Any feedback appreciated

Cheers
Ronny
